cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceQueryAdmin.java

```diff
@@ -103,11 +102,6 @@ public NamespaceMeta get(Id.Namespace namespaceId) throws Exception {
    */
   @Override
   public boolean exists(Id.Namespace namespaceId) throws Exception {
-    try {
-      get(namespaceId);
-    } catch (NotFoundException e) {
-      return false;
-    }
-    return true;
+    return nsStore.get(namespaceId) != null;
```

Comment:
Yes, but I was wondering if ``exists`` should just throw ``UnsupportedOperationException`` in the Http Client. But that would be bad because its a public client.

Re: the other security breach you mentioned, had a discussion about that with @anew. The policy in CDAP right now is - The list APIs will not show you entities that you do not have access to. However, if by some way you are able to get the identifier for the entity (say namespace + app), you can tell if the entity exists or not. The reason was that existence checks are also done as part of other operations - e.g. creating a dataset first checks if the dataset exists - if that check returns false, it creates a dataset. If we added an authorization check before the existence check, that operation will never succeed in such scenarios. This is documented at https://wiki.cask.co/display/CE/Authorization+-+CDAP+3.5#Authorization-CDAP3.5-AuthorizationPolicies, but we can revisit as a whole later too.